flink-clients/src/test/java/org/apache/flink/client/deployment/application/ClassPathPackagedProgramRetrieverTest.java

```diff
@@ -285,25 +148,55 @@ public void testJobGraphRetrievalFailIfDoesNotFindTheEntryClassInTheJobDir()
     }
 
     @Test
-    public void testRetrieveCorrectUserClasspathsWithoutSpecifiedEntryClass()
+    public void testSavepointRestoreSettings()
+            throws FlinkException, IOException, ProgramInvocationException {
+        final Configuration configuration = new Configuration();
+        final SavepointRestoreSettings savepointRestoreSettings =
+                SavepointRestoreSettings.forPath("foobar", true);
+        final JobID jobId = new JobID();
+
+        configuration.setString(PipelineOptionsInternal.PIPELINE_FIXED_JOB_ID, jobId.toHexString());
+        SavepointRestoreSettings.toConfiguration(savepointRestoreSettings, configuration);
+
+        final PackagedProgramRetriever retrieverUnderTest =
+                newBuilder(PROGRAM_ARGUMENTS)
+                        .setJobClassName(TestJob.class.getCanonicalName())
+                        .build();
+
+        final JobGraph jobGraph = retrieveJobGraph(retrieverUnderTest, configuration);
+
+        assertThat(jobGraph.getSavepointRestoreSettings(), is(equalTo(savepointRestoreSettings)));
+        assertEquals(jobGraph.getJobID(), jobId);
+    }
+
+    @Test
+    public void testJarFromClassPathSupplierSanityCheck() {
+        Iterable<File> jarFiles = JarsOnClassPath.INSTANCE.get();
+
+        // Junit executes this test, so it should be returned as part of JARs on the class path
+        assertThat(jarFiles, hasItem(hasProperty("name", containsString("junit"))));
+    }
+
+    @Test
+    public void testRetrieveCorrectUserClassPathsWithoutSpecifiedEntryClass()
             throws IOException, FlinkException, ProgramInvocationException {
-        final ClassPathPackagedProgramRetriever retrieverUnderTest =
-                ClassPathPackagedProgramRetriever.newBuilder(PROGRAM_ARGUMENTS)
+        final PackagedProgramRetriever retrieverUnderTest =
+                newBuilder(PROGRAM_ARGUMENTS)
                         .setJarsOnClassPath(Collections::emptyList)
                         .setUserLibDirectory(userDirHasEntryClass)
                         .build();
         final JobGraph jobGraph = retrieveJobGraph(retrieverUnderTest, new Configuration());
 
         assertThat(
                 jobGraph.getClasspaths().stream().map(URL::toString).collect(Collectors.toList()),
-                containsInAnyOrder(expectedURLs.stream().map(URL::toString).toArray()));
+                containsInAnyOrder(EXPECTED_URLS.stream().map(URL::toString).toArray()));
     }
 
     @Test
-    public void testRetrieveCorrectUserClasspathsWithSpecifiedEntryClass()
+    public void testRetrieveCorrectUserClassPathsWithSpecifiedEntryClass()
```

Comment:
Correct me if I'm wrong here but don't we want to move this into `PackaedProgramRetrieverTestBase` as well?